services/transactionService.ts

```diff
@@ -163,45 +113,26 @@ export async function upsertManyTransactionsForAddress (transactions: Transactio
   return ret.filter((t) => t !== undefined) as TransactionWithAddressAndPrices[]
 }
 
-export async function syncTransactionsForAddress (addressString: string): Promise<TransactionWithAddressAndPrices[]> {
+export async function syncTransactionsForAddress (addressString: string, start: number = 0, maxTransactions: number = Infinity): Promise<TransactionWithAddressAndPrices[]> {
   const address = await fetchAddressBySubstring(addressString)
-  let newTransactionsCount = -1
-  let seenTransactionsCount = 0
-  let insertedTransactions: TransactionWithAddressAndPrices[] = []
-  while (newTransactionsCount !== 0) {
-    const addressTransactions = await getAddress({
-      address: address.address,
-      nbFetch: FETCH_N,
-      nbSkip: seenTransactionsCount
-    })
-    let nextConfirmedTransactions = addressTransactions.confirmedTransactionsList
-    let nextUnconfirmedTransactions = addressTransactions.unconfirmedTransactionsList.map(mempoolTx => parseMempoolTx(mempoolTx))
-
-    // remove transactions older than the networks
-    nextConfirmedTransactions = nextConfirmedTransactions.filter(txThesholdFilter(address))
-    nextUnconfirmedTransactions = nextUnconfirmedTransactions.filter(txThesholdFilter(address))
 
-    newTransactionsCount = nextConfirmedTransactions.length // Unconfirmed transactions are NOT taken into account when skipping with nbSkip
-    seenTransactionsCount += newTransactionsCount
+  const addressTransactions = await getAddressTransactions(addressString, start, maxTransactions)
+  const insertedTransactions: TransactionWithAddressAndPrices[] = [
+    ...await upsertManyTransactionsForAddress(addressTransactions.confirmed, address),
+    ...await upsertManyTransactionsForAddress(addressTransactions.unconfirmed, address)
+  ]
 
-    let newInsertedTransactions = await upsertManyTransactionsForAddress(nextConfirmedTransactions, address)
-    newInsertedTransactions = [
-      ...newInsertedTransactions,
-      ...(await upsertManyTransactionsForAddress(nextUnconfirmedTransactions, address, false))
-    ]
-    insertedTransactions = [...insertedTransactions, ...newInsertedTransactions]
-    await new Promise(resolve => setTimeout(resolve, FETCH_DELAY))
-  }
+  if (maxTransactions === Infinity || addressTransactions.confirmed.length < maxTransactions) { await updateLastSynced(addressString) } else { void syncTransactionsForAddress(addressString, start + addressTransactions.confirmed.length) }
 
-  await updateLastSynced(addressString)
   return insertedTransactions
 }
 
-export async function syncTransactionsAndPricesForAddress (addressString: string): Promise<void> {
+export async function syncTransactionsAndPricesForAddress (addressString: string, maxTransactions: number = DEFAULT_NUMBER_OF_TRANSACTIONS_TO_SYNC_INITIALLY): Promise<TransactionWithAddressAndPrices[]> {
```

Comment:
1. Create an interface just for 2 parameters, isn't that overkill? 
2. About not having a default value, I agree, makes every call more explicit and easy to understand

Reply:
1. Idd, I though this was like the other place I mentioned that also had the `start` parameter. But it could have tho, right? Instead of always calling the other function with 0

Reply:
It would make sense for me to have `start` here if we had a use case to sync only from a number X of transactions backwards in time, but that wouldn't make sense as, when we call a sync, we don't and can't know how many new transactions this address had since the last sync, so every sync will always start with 0.